src/lightning/fabric/fabric.py

```diff
@@ -404,22 +409,36 @@ def backward(self, tensor: Tensor, *args: Any, model: Optional[_FabricModule] =
             model as argument here.
 
         """
-        module = model._forward_module if model is not None else model
-        if isinstance(self._strategy, DeepSpeedStrategy):
-            if model is None:
-                if self._models_setup == 0:
-                    raise RuntimeError("No models were set up for backward. Did you forget to call `fabric.setup()`?")
-                if self._models_setup > 1:
-                    raise ValueError(
-                        "When using multiple models + deepspeed, please provide the model used to perform"
-                        " the optimization: `self.backward(loss, model=model)`"
-                    )
-                module = self._strategy.model
-            else:
-                # requires to attach the current `DeepSpeedEngine` for the `_FabricOptimizer.step` call.
-                self._strategy._deepspeed_engine = module
-
-        self._strategy.backward(tensor, module, *args, **kwargs)
+        if _is_lightning_module:  # cannot use `isinstance` because fabric doesnt install pytorch lightning
+            # old code path: the LightningModule doesn't use `_BackwardTensor`
+            module = model._forward_module if model is not None else model
+            if isinstance(self._strategy, DeepSpeedStrategy):
+                if model is None:
+                    if self._models_setup == 0:
+                        raise RuntimeError(
+                            "No models were set up for backward. Did you forget to call `fabric.setup()`?"
+                        )
+                    if self._models_setup > 1:
+                        raise ValueError(
+                            "When using multiple models + deepspeed, please provide the model used to perform"
+                            " the optimization: `self.backward(loss, model=model)`"
+                        )
+                    module = self._strategy.model
+                else:
+                    # requires to attach the current `DeepSpeedEngine` for the `_FabricOptimizer.step` call.
+                    self._strategy._deepspeed_engine = module
+            self._strategy.backward(tensor, module, *args, **kwargs)
+            return None
```

Comment:
This block is kept to preserve support for wrapping a `LightningModule` with `Fabric`.

I don't think there's a reliable way of using `_BackwardTensor` in the `LightningModule`, and if there was, it would only be useful in the manual optimization case to avoid `self.manual_backward`. It's not worth it.